Newsletters: add 241 (2023-03-08)

harding opened the pull request:
Also adds a blog post announcing the podcast.

- [x] Lede, alt-vault proposal, releases/RCs, topic links
- [x] Bitcoin Core PR Review Club summary @LarryRuane 
- [x] LND 7462 @adamjonas

paulkania commented on _posts/en/2023-03-08-podcast.md:
```diff
@@ -0,0 +1,67 @@
+New podcast episodes will be released a few days after the conclusion of
+a Twitter Spaces to allow for professional post production of the audio.
+Transcripts will be published a few days after that.  In addition to
+publishing new episodes as soon as they've complete, Optech's audio and
```
"they've complete," ought to be "they've completed," of "they are complete,"

paulkania commented on _posts/en/2023-03-08-podcast.md:
```diff
@@ -0,0 +1,67 @@
+
+If you're interested in Bitcoin technology, we encourage you to
+subscribe to the podcast and consider joining us each week for a new
+recap on Twitter Spaces.
```
Maybe add the time when Twitter Spaces are held here?

harding replied:
I added 15:00 Thursday to the blog post in a comment.  I'll leave it up to @bitschmidty to decide whether or not to publish it with or without changes.

bitschmidty commented on _posts/en/2023-03-08-podcast.md:
```diff
@@ -0,0 +1,67 @@
+  Recap hosted on Twitter Spaces.
+
+---
+Since FIXME, Optech contributors Mike Schmidt and Mark "Murch" Erhardt
```
Can be either Newsletter 210 and/or July 27 2022

bitschmidty commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,147 @@
+      tapscript], one with her unvault tapscript and one with her freeze
+      tapscript, e.g. `tr(key,{trigger,freeze})`.
+
+      - The *trigger tapscript* contains her less-trusted authorization
```
Im still green on this proposal, but above we say "two tapscripts, one with her **unvault** tapscript and one with her **freeze** tapscript" but here we say "trigger" and "freeze" tapscripts

harding replied:
Good catch, thanks!  I rephrased several places to be more consistent, which I think actually simplified things and made them more clear.  Thanks!

bitschmidty commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,147 @@
+        parameter, e.g. a relative timelock of 1,000 blocks (about 1
+        week).
+
+      - The *freeze tapscript* contains any authorization conditions
```
Does it make sense to note here that the freeze doesn’t "require" a spend delay? (as I think this is implied when we note later in the writeup that Alice can spend using the freeze tapscript branch if she notices shenanigans)

harding replied:
Add mention about time delays (or lack thereof) to preceding paragraph and also added several other time mentions throughout the text to help clarify this point.  Thanks!

bitschmidty commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,147 @@
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [Core Lightning 23.02rc3][] is a release candidate for a new
```
https://github.com/ElementsProject/lightning/releases/tag/v23.02

bitschmidty reviewed (commented):
Initial comments

glozow commented on _posts/en/2023-03-08-podcast.md:
```diff
@@ -0,0 +1,67 @@
+Years from now, users, developers, and researchers will be able to look
+up a subject in Optech's extensive [topic index][], choose a particular
+item of interest, review a brief summary of the event in the newsletter,
+and then listen or read contemporary discussion about the subject from
```
```suggestion
and then listen to or read contemporary discussion about the subject from
```

glozow commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,150 @@
+      either the vaulting or initial unvaulting transactions.  Recall
+      that Alice explicitly placed the freeze tapscript in the vaulting
+      transaction, and note that it was implicitly carried over by the
+      transaction with initiated the unvaulting.
```
```suggestion
      transaction that initiated the unvaulting.
```

bitschmidty commented on _posts/en/2023-03-08-podcast.md:
```diff
@@ -0,0 +1,67 @@
+
+Each episode will also be available as a page on the Optech website with
+a summary of the episode content and a complete transcript.  We think
+this is major advance in Optech's mission to improve communication
```
```suggestion
this is a major advance in Optech's mission to improve communication
```

bitschmidty commented:
[This](https://github.com/lightningnetwork/lnd/issues/5594#issuecomment-1453908298) piece of news was passed to me for consideration in the newsletter. Im not sure we cover roadmaps on their own, but I suppose if there is a related LND/bolt12 update it could be worked in.

LarryRuane commented:
Added review club summary 8122c2f707beb17e7ddb314d18d98a299b61b371

glozow commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,209 @@
+question below to see a summary of the answer from the meeting.*
+
+[bitcoin-inquisition: Activation logic for testing consensus changes][review club bi-16]
+is a PR by Anthony Towns that improves activation and deactivation
+logic for the [Bitcoin Inquisition][] project, which runs on [signet][topic signet].
```
I quite strongly think "improves" is not the right word here. It swaps it out for a different mechanism; they are used in different contexts with very different goals. I don't think there is any notion of one being better than the other.
```suggestion
is a PR by Anthony Towns that adds a new method for activating and deactivating
soft forks in the [Bitcoin Inquisition][] project, designed to be run on [signet][topic signet]
and used for testing.
```

glozow commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,209 @@
+logic for the [Bitcoin Inquisition][] project, which runs on [signet][topic signet].
+This project was covered in [newsletter #219 bi][].
+
+While most of the [Bitcoin Inquisition PRs][bi prs] implement specific consensus or
+relay policy changes, this PR modifies the Bitcoin Inquisition framework itself.
```
```suggestion
```
It's unclear what "the Bitcoin Inquisition framework" would mean.

glozow commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,209 @@
+mechanism -- on a test network activating these changes can be streamlined.
+The PR also implements a way to deactivate changes that turn out to be buggy
+or undesired, which is a major departure from mainnet.
+Specifically, this PR replaces [BIP9][] block version bit semantics with what
+are called [Heretical Deployments][].
```
I think the paragraph would flow better if this was the first sentence, rather than the last sentence.

glozow commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,209 @@
+      running, so even after a bug has been fixed, some users may continue
+      running the buggy version. Depending only on regtest makes
+      integration testing third-party software more difficult.
+      The threshold of merging consensus changes to a signet-only repository
+      is much lower than merging to Core.
+      Merging changes to Core that aren't yet activated introduces the
+      risk that bugs can leak in a way that affects existing behavior."
```
```suggestion
      Merging consensus changes to a separate repository is much less risky than merging to Core;
      adding soft fork logic, even if not activated, may introduce bugs that affect existing behavior."
```

glozow commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,209 @@
+      machine period (this period is fixed for Heretical Deployment)."
+  a2link="https://bitcoincore.reviews/bitcoin-inquisition-16#l-126"
+
+  q3="Why is Taproot buried in signet?"
```
```suggestion
  q3="Why is Taproot buried in this PR?"
```

LarryRuane commented:
Force pushed to address @glozow's suggestions -- thanks!

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+  [Newsletter #234][news234 vault]).  His alternative would add three
+  opcodes instead of two.  To provide an example:
+
+    - *Alice deposits funds in a vault* by paying a [taproot][topic
+      taproot] tree which contains at least two [tapscripts][topic
+      tapscript], one with her unvault tapscript and one with her freeze
+      tapscript, e.g. `tr(key,{trigger,freeze})`.
```
Sorry, opinionated on the terminology here. ;)
I would prefer “(taproot) script tree” over “taproot tree”. I would prefer referring to the scripts in the script tree as “leafscripts”, from what I understand “Tapscript” refers to the specific set of Script rules defined by BIP342 to be used in leafscripts with leaf version 0xc0, not a leafscript that uses those rules.

```suggestion
    - *Alice deposits funds in a vault* by paying a P2TR output with a [script tree][topic
      taproot] which contains at least two [leafscripts][topic
      tapscript], one with her unvault leafscript and one with her freeze
      leafscript, e.g. `tr(key,{trigger,freeze})`.
```

harding replied:
> I would prefer “(taproot) script tree” over “taproot tree”.

I'm fine with this.

> I would prefer referring to the scripts in the script tree as “leafscripts”, from what I understand “Tapscript” refers to the specific set of Script rules defined by BIP342 to be used in leafscripts with leaf version 0xc0, not a leafscript that uses those rules.

I can't think of a good argument against this (and, yeah, I tried :-), so I'll make these changes, but since you have a strong opinion about this, would you consider opening a PR to edit https://bitcoinops.org/en/topics/tapscript/ to include a description of the differences between tapscript and leafscript (and adding an alias in the YAML so that we get leafscript in our topics index)?  If you're willing, you can do this at your leisure; it's not blocking on merging this newsletter.  Please CC me if you open a PR as I don't subscribe to all PRs on the repo.

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+      tapscript], one with her unvault tapscript and one with her freeze
+      tapscript, e.g. `tr(key,{trigger,freeze})`.
+
+      - The *trigger tapscript* contains her less-trusted authorization
+        conditions (such as requiring a signature from her hot wallet)
+        and an `OP_TRIGGER_FORWARD` opcode.  At the time she creates
+        this tapscript, she provides the opcode a *spend delay*
+        parameter, e.g. a relative timelock of 1,000 blocks (about 1
+        week).
```
```suggestion
      - The *trigger leafscript* contains her less-trusted authorization
        conditions (such as requiring a signature from her hot wallet)
        and an `OP_TRIGGER_FORWARD` opcode.  At the time she creates
        this leafscript, she provides the opcode a *spend delay*
        parameter, e.g. a relative timelock of 1,000 blocks (about 1
        week).
```

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+        parameter, e.g. a relative timelock of 1,000 blocks (about 1
+        week).
+
+      - The *freeze tapscript* contains any authorization conditions
+        Alice wants to specify (including none at all) and
+        an `OP_FORWARD_DESTINATION` opcode.  At the time she creates
+        this tapscript, she also chooses her more-trusted authorization
+        conditions (such as requiring multiple signatures from multiple
+        cold wallets and hardware signing devices).  She provides the
+        opcode a commitment to those conditions in the form of a hash
+        digest.
```
```suggestion
      - The *freeze leafscript* contains any authorization conditions
        Alice wants to specify (including none at all) and
        an `OP_FORWARD_DESTINATION` opcode.  At the time she creates
        this leafscript, she also chooses her more-trusted authorization
        conditions (such as requiring multiple signatures from multiple
        cold wallets and hardware signing devices).  She provides the
        opcode a commitment to those conditions in the form of a hash
        digest.
```

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+        digest.
+
+    - *Alice initiates an unvaulting* by spending the output received to
+      the above taproot tree (using it as an input) and choosing the
```
Since the script tree is only a subset of the spending conditions in a P2TR output, the “received to the taproot tree” feels a bit odd to me. 

```suggestion
      the output with the above script tree (using it as an input) and choosing the
```

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+
+    - *Alice initiates an unvaulting* by spending the output received to
+      the above taproot tree (using it as an input) and choosing the
+      trigger tapscript.  At this time, she provides two additional
```
```suggestion
      trigger leafscript.  At this time, she provides two additional
```

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+      output which will receive this input's funds and a hash-based
+      commitment to how she wants to be able to spend the funds later.
+      The opcode verifies that the indicated output of this transaction
+      pays a taproot tree similar to the one being spent except that the
+      trigger tapscript is replaced with a script using an
```
```suggestion
      pays a P2TR output with a script tree similar to the one being spent except that the
      trigger leafscript is replaced with a script using an
```

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+      `OP_CHECKSEQUENCEVERIFY` (CSV) relative delay equal to the delay
+      specified previously (e.g., 1000 blocks) and an
+      `OP_FORWARD_OUTPUTS` opcode which includes Alice's commitment
+      hash.  The method of reconstructing the taproot tree is similar to
+      an earlier [covenant][topic covenants] proposal,
```
```suggestion
      hash.  The method of reconstructing the script tree is similar to
      an earlier [covenant][topic covenants] proposal,
```

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+      timelock has expired and then spending the unvaulting output,
+      choosing the tapleaf with the `OP_FORWARD_OUTPUTS` opcode.  The
+      opcode verifies that the spending transaction's output amounts and
+      scripts hash to the same commitment Alice made in the previous
```
```suggestion
      script’s hash to the same commitment Alice made in the previous
```

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+      did want to spend the funds to the specified outputs, and
+      completed the spend.
+
+    - If something goes wrong, *Alice freezes the funds*.  She can do
+      this at any time up until an unvaulting is completed by simply
+      choosing to spend using the freeze tapscript from the output of
+      either the vaulting or initial unvaulting transactions.  Recall
+      that Alice explicitly placed the freeze tapscript in the vaulting
+      transaction, and note that it was implicitly carried over by the
+      transaction with initiated the unvaulting.
```
```suggestion
    - If something goes wrong, *Alice freezes the funds*.  She can do
      this at any time up until an unvaulting is completed by simply
      choosing to spend using the freeze leafscript from the output of
      either the vaulting or initial unvaulting transactions.  Recall
      that Alice explicitly placed the freeze leafscript in the vaulting
      transaction, and note that it was implicitly carried over by the
      transaction with initiated the unvaulting.
```

Xekyo commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+      transaction, and note that it was implicitly carried over by the
+      transaction with initiated the unvaulting.
+
+  One of the advantages to users of this approach over the original
+  `OP_VAULT` design is that the freeze tapscript can contain any
+  authorization conditions Alice wants to specify.  In the `OP_VAULT`
+  proposal, anyone knowing the parameters chosen by Alice could spend
+  her funds to the freeze script.  That wasn't a security problem but it
+  could be annoying.  In Sanders's design, Alice could (for example)
+  require a signature from a very lightly protected wallet in order to
+  initiate a freeze---this would perhaps be enough of a burden to
+  prevent most griefing attacks but not enough of a barrier to prevent
+  Alice from quickly freezing her funds in an emergency.
```
```suggestion
  One of the advantages to users of this approach over the original
  `OP_VAULT` design is that the freeze leafscript can contain any
  authorization conditions Alice wants to specify.  In the `OP_VAULT`
  proposal, anyone knowing the parameters chosen by Alice could spend
  her funds to the freeze script.  That wasn't a security problem but it
  could be annoying.  In Sanders's design, Alice could (for example)
  require a signature from a very lightly protected wallet in order to
  initiate a freeze---this would perhaps be enough of a burden to
  prevent most griefing attacks but not enough of a barrier to prevent
  Alice from quickly freezing her funds in an emergency.
```

Xekyo reviewed (commented):
Really liked the write-up of Greg’s idea. Curious on your thoughts regarding terminology.

harding commented:
Pushed edits for all reviewer feedback, thanks @paulkania @Xekyo @bitschmidty @glozow !  Added lede, updated releases/RCs, and added topic links.  Reviewed contributions by @LarryRuane and @adamjonas (thanks!).  Thanks everyone!

bitschmidty reviewed (approved):
Reviewed releases section, topic links, PR Review Club (2 small changes made), and ensured all review comments were resolved.

Squashed commits.

bitschmidty commented:
Thank you @harding for authoring the newsletter (and blog post announcement!) with  @adamjonas @LarryRuane! And thanks to @glozow @paulkania @Xekyo for the reviews 🚀

jamesob commented on _posts/en/newsletters/2023-03-08-newsletter.md:
```diff
@@ -0,0 +1,226 @@
+
+  One of the advantages to users of this approach over the original
+  `OP_VAULT` design is that the freeze leafscript can contain any
+  authorization conditions Alice wants to specify.  In the `OP_VAULT`
```
For what it's worth, this was the case with BIP 345 before Sanders made these suggestions. Search for [`optional auth. scriptPubKey](https://github.com/jamesob/bips/blob/jamesob-23-02-opvault/bip-vaults.mediawiki#user-content-Parameters) in the current text of the BIP. Greg's rework doesn't add this in.

harding replied:
Was that the result of some previous change in the design?  In our original coverage of the idea, we linked to a post by AJ where he mentioned that as a concern, see https://bitcoinops.org/en/newsletters/2023/01/18/#proposal-for-new-vault-specific-opcodes where we wrote,

> Towns [notes](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2023-January/021328.html) that [...] any third party who learns the address can also freeze the user’s funds (although they’ll have to pay a transaction fee to do so), creating an inconvenience for the user.

If it was a previous change to the proposal that added the ability to allow requiring arbitrary authorization for freezing, then I think we're technically correct to say the an advantage of Greg's approach over the original approach (as posted to the ML on Jan 9th) is this ability.

However, I also get that it's misleading in the sense that it's not an advantage over what you wanted to ship at the time Greg suggested it.  Would you like us to print a correction in next week's newsletter?
